subtests/docker_cli/systemd/systemd.py

```diff
@@ -1,86 +1,155 @@
+        img_lst = ''
+        for img in DockerImages(self).list_imgs_full_name():
+            img_lst = '{} {}'.format(img_lst, img.split('/', -1)[-1])
+        if self.config['image_name'] not in img_lst:
+            raise DockerTestError("Image %s is not present"
+                                  % self.config['image_name'])
 
     def cleanup(self):
-        super(systemd, self).cleanup()
-        filepath = self.config['sysd_unitf_dest']
+        super(systemd_base, self).cleanup()
+        filepath = self.sub_stuff['unitf_dst']
         if exists(filepath):
             self.sysd_action('stop', self.config['sysd_unit_file'])
             unlink(filepath)
             self.sysd_action('daemon-reload')
-        DockerImages(self).clean_all([self.config['name']])
+        DockerImages(self).clean_all([self.config['image_name']])
 
     @staticmethod
-    def sysd_action(action=None, sysd_unit=None):
+    def sysd_action(action, sysd_unit=None):
         command = 'systemctl {}'.format(action)
         if sysd_unit:
             command = '{} {}'.format(command, sysd_unit)
         utils.run(command, ignore_status=False)
 
-    @staticmethod
-    def read_socket(host, port):
-        max_read_tries = 10
-        num_read_try = max_read_tries
-        while num_read_try:
-            # using timeout 13-sec as it's 1-sec longer than two DNS timeouts
-            s = socket.create_connection((host, port), 13)
-            s.settimeout(2.0)
-            try:
-                time_from_socket = s.recv(512).strip()
-                if time_from_socket:
-                    s.shutdown(socket.SHUT_RDWR)
-                    return time_from_socket
-                num_read_try -= 1
-            except socket.timeout:
-                pass
-            time.sleep(1.0)
-        raise DockerTestError("Failed to read from socket with %d tries"
-                              % max_read_tries)
+    def sed_file(self, vars_dict):
+        tmpdirf = self.sub_stuff['tmpdirf']
+        tmp_write = '{}.tmp'.format(tmpdirf)
+        with open(tmpdirf, "r") as srcfile, open(tmp_write, "w") as dstfile:
+            dstfile.write(srcfile.read().format(**vars_dict))
+        rename(tmp_write, tmpdirf)
+
+# r"""
+# Summary
+# -------
+#
+# To test building an image using systemd unitfile
+#
+# Operational Summary
+# ---------------------
+#
+# #. Edit unitfile ``docker-build.service`` & copy it to /etc/systemd/system
+# #. Edit Dockerfile
+# #. Systemd service is started using this file to build an image.
+# #. Systemd service is stoped and image removed
+# """
+
+
+class systemd_build(systemd_base):
+
+    def initialize(self):
+        # update dict, edit unitfile & copy it to /etc/systemd/system/
+        super(systemd_build, self).initialize()
+        build_dict = {'dockerfiledir': self.sub_stuff['bindir'],
+                      'options': self.config['unit_opts']}
+        vars_dict = self.sub_stuff['vars_dict']
```

Comment:
@edsantiago super has:
```self.sub_stuff['vars_dict'] = {'name': self.config['image_name']}```
But I think I should have the classes set their own instead?